### PR TITLE
PySide2 missing ./plugins/xcbglintegrations/libqxcb-glx-integration.so in linux.

### PR DESCRIPTION
**Pull request by mjscosta**
pyinstaller crawls for the dependencies, it does not find or set the opengl dependencies required for PySide2, giving this error:

`QXcbIntegration: Cannot create platform OpenGL context, neither GLX nor EGL are enabled`

required lib ./plugins/xcbglintegrations/libqxcb-glx-integration.so not copied to qt5_plugins folder.

relates to this issue #1930 , but its a fix for PySide2 only, as it is the issue I need to solve now.

### CODE REVIEW

**Comment by mjscosta** [comment]
Seemed to have failed due to some errors on dev.

**Comment by tpeeters** [comment]
Is this being reviewed?

I needed to do the same change to make PyInstaller work with PySide2 on Ubuntu 16.04 with Python 3.6.

File: PyInstaller/loader/rthooks/pyi_rth_qt5plugins.py [hunk 1]
```diff
@@ -50,3 +50,9 @@
 
 # We set "qt5_plugins" as only one path for Qt5 plugins
 QCoreApplication.setLibraryPaths([os.path.abspath(d)])
+
+# Fix some Qt5 errors in linux
+if sys.platform.startswith('linux'):
+    os.environ['FONTCONFIG_FILE'] = '/etc/fonts/fonts.conf'  # Address missing font configuration issue
+    os.environ['FONTCONFIG_PATH'] = '/etc/fonts/'
+    os.environ['QT_XKB_CONFIG_ROOT'] = '/usr/share/X11/xkb'  # Address keyboard input issue
```
Comment (htgoebel) [inline PyInstaller/loader/rthooks/pyi_rth_qt5plugins.py]: I would expect this to be the defaults in Qt anyway. Why is this necessary?

File: PyInstaller/__init__.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 # Note: Keep this variable as plain string so it could be updated automatically
 #       when doing a release.
-__version__ = '3.4.dev0'
+__version__ = '3.4.dev5'
```
Comment (htgoebel) [inline PyInstaller/__init__.py]: Please don't change the version number in a pull-request.

File: bootloader/src/pyi_utils.c [hunk 3]
```diff
@@ -221,7 +221,7 @@ pyi_unsetenv(const char *variable)
     #if HAVE_UNSETENV
     rc = unsetenv(variable);
     #else /* HAVE_UNSETENV */
-    rc = setenv(variable, "");
+    rc = setenv(variable, "", true);
```
Comment (htgoebel) [inline bootloader/src/pyi_utils.c]: Reading ht man-page for `setnv` I have the impression you spotted a long standing bug. Well done!
Unfortunately this change is unrelated to this pull-request. Please provide an extra pull-request. Thanks

**Comment by bjones1** [comment]
I'd like to merge #3693 first, That might actually fix part of (or all of) this, since the Qt code should automatically find a lot of the XCB stuff -- see https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/utils/hooks/qt.py#L399.

File: PyInstaller/loader/rthooks/pyi_rth_qt5plugins.py (hunk 1 above)
Comment (mjscosta) [reply PyInstaller/loader/rthooks/pyi_rth_qt5plugins.py]: In ubuntu 16.04, using PySide2 from conda forge, it uses qt 5.6.2 from anaconda default channel.
Both fonts and keyboard input do not work in the packaged pyinstaller executable. I don't know about other platforms or qt versions, if it still is an issue. For reference check: https://tutel.me/c/programming/questions/42595685/this+application+failed+to+start+because+it+could+not+find+or+load+the+qt+platform+plugin+quotxcbquot+in+quotquot.

File: PyInstaller/__init__.py (hunk 2 above)
Comment (mjscosta) [reply PyInstaller/__init__.py]: Right, sorry about that, should have created a branch for the pull request. I've been pushing changes that I need, and building my own pyinstaller version. I'll try to fix that asap.

File: bootloader/src/pyi_utils.c (hunk 3 above)
Comment (mjscosta) [reply bootloader/src/pyi_utils.c]: Ok. I've added that to solve a compile time error, and checking the code the change made some sense.

**Comment by mjscosta** [comment]
Great, I'll make the changes as soon as possible, probably I be able to get it done by the end of the week. Afterworlds, I can test it.

File: PyInstaller/loader/rthooks/pyi_rth_qt5plugins.py (hunk 1 above)
Comment (mjscosta) [reply PyInstaller/loader/rthooks/pyi_rth_qt5plugins.py]: Adding those solved the issue, i'd appreciate some feedback, on if its a version specific issue, and if that's the case, I can remove it.

**Comment by mjscosta** [comment]
I still haven't tested Windows, so I can not tell anything about #3568.

File: PyInstaller/loader/rthooks/pyi_rth_qt5plugins.py (hunk 1 above)
Comment (mjscosta) [reply PyInstaller/loader/rthooks/pyi_rth_qt5plugins.py]: please refer to https://github.com/pyinstaller/pyinstaller/issues/3724

**Comment by mjscosta** [comment]
@bjones1, you can drop this pull request, I've tested with the latest develop branch, and that problem seems to be solved. Mostly this reference you added https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/utils/hooks/qt.py#L399, seems to solve it.

The remaining issues, I've spited into their own pull request.

**Comment by bjones1** [comment]
OK, thanks for testing!
